File: software/examples/c/example_simple.c
#include <stdio.h>

#include "ip_connection.h"
#include "bricklet_gps.h"

#define HOST "localhost"
#define PORT 4223
#define UID "XYZ" // Change to your UID

int main(void) {
	// Create IP connection
	IPConnection ipcon;
	ipcon_create(&ipcon);

	// Create device object
	GPS gps;
	gps_create(&gps, UID, &ipcon);

	// Connect to brickd
	if(ipcon_connect(&ipcon, HOST, PORT) < 0) {
		fprintf(stderr, "Could not connect\n");
		return 1;
	}
	// Don't use device before ipcon is connected

	// Get current coordinates
	uint32_t latitude, longitude;
	char ns, ew;
	uint16_t pdop, hdop, vdop, epe;
	if(gps_get_coordinates(&gps, &latitude, &ns, &longitude, &ew,
	                       &pdop, &hdop, &vdop, &epe) < 0) {
		fprintf(stderr, "Could not get coordinates, probably timeout\n");
		return 1;
	}

	printf("Latitude: %f° %c\n", latitude/1000000.0, ns);
	printf("Longiutde: %f° %c\n", longitude/1000000.0, ew);

	printf("Press key to exit\n");
	getchar();
	ipcon_destroy(&ipcon); // Calls ipcon_disconnect internally
	return 0;
}

File: software/examples/go/example_simple.go
package main

import (
	"fmt"
	"github.com/Tinkerforge/go-api-bindings/gps_bricklet"
	"github.com/Tinkerforge/go-api-bindings/ipconnection"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your GPS Bricklet.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	gps, _ := gps_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	// Get current coordinates.
	latitude, ns, longitude, ew, _, _, _, _, _ := gps.GetCoordinates()

	fmt.Printf("Latitude: %f °\n", float64(latitude)/1000000.0)
	fmt.Printf("N/S: \n", ns)
	fmt.Printf("Longitude: %f °\n", float64(longitude)/1000000.0)
	fmt.Printf("E/W: \n", ew)

	fmt.Print("Press enter to exit.")
	fmt.Scanln()

}
